### DISCUSSION TITLE
ref Switch Expression

### DISCUSSION THREAD

**Discussion by ilyfairy** [dotnet/csharplang#6757] [Language Ideas] [answered]
Make Switch expression support ref
Now it prompts an error "Invalid expression term \'ref\'"

```CSharp
string str = "a1";
int x = 1;
int y = 2;

ref int r = ref str switch
{
    "a1" => ref x,
    "a2" => ref y,
    _ => ref Unsafe.NullRef<int>()
};
```

**Comment by bernd5** [comment]
Today you can do something like that with conditional expressions, only.

For example:
```CSharp
using System;
using System.Runtime.CompilerServices;

string str = "a1";
int x = 1;
int y = 2;

ref int r = 
      ref str is "a1" ? ref x 
    : ref str is "a2" ? ref y
    : ref Unsafe.NullRef<int>();

//see if we have a ref...
r++;
r++;
r++;
Console.WriteLine(x);
```

In my eyes it would be great if we could use switch expressions here, too.

See: https://github.com/dotnet/csharplang/blob/3c8559f186d4c5df5d1299b0eaa4e139ae130ab6/meetings/2018/LDM-2018-04-04.md#reflvalue-producing-pattern-switch-expression

**Comment by 333fred** [comment, marked as answer]
Duplicate of https://github.com/dotnet/csharplang/issues/3326. Note that this is an Any Time proposal, meaning that it is open for community contribution.
